src/Messenger/UpdateHandler.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+/*
+ * This file is part of the Mercure Component project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Symfony\Component\Mercure\Messenger;
+
+use Symfony\Component\Mercure\PublisherInterface;
+use Symfony\Component\Mercure\Update;
+
+/**
+ * @author Saif Eddin Gmati <azjezz@protonmail.com>
+ *
+ * @experimental
+ */
+final class UpdateHandler
+{
+    private $publisher;
+
+    public function __construct(PublisherInterface $publisher)
+    {
+        $this->publisher = $publisher;
+    }
+
+    public function __invoke(Update $update): void
+    {
+        $this->publisher->publish($update);
```

Comment:
Adding a `method_exists()` check falling back to `__invoke()` here would avoid the need for doing such check at compile time in mercure-bundle (which is responsible for registering this class as a `messenger.message_handler` tagged service and will need to be patched according to this PR).